diff --git a/src/rand/sysrng.c b/src/rand/sysrng.c
index 5a92114..36ad015 100644
--- a/src/rand/sysrng.c
+++ b/src/rand/sysrng.c
@@ -189,6 +189,15 @@ seeder_win32(const br_prng_class **ctx)
 }
 #endif
 
+unsigned long genrand_int32(void);
+
+static int
+seeder_mtrand(const br_prng_class **ctx)
+{
+
+	return genrand_int32();
+}
+
 /*
  * An aggregate seeder that uses RDRAND, and falls back to an OS-provided
  * source if RDRAND fails.
@@ -245,8 +254,8 @@ br_prng_seeder_system(const char **name)
 	return &seeder_win32;
 #else
 	if (name != NULL) {
-		*name = "none";
+		*name = "mt";
 	}
-	return 0;
+	return  &seeder_mtrand;
 #endif
 }
